src/libmeasurement_kit/ooni/ooni_reporter.cpp

```diff
@@ -0,0 +1,89 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+OoniReporter::OoniReporter(const OoniTest &parent) {
+    settings = parent.options; // Copy the parent's settings
+    logger = parent.logger;
+    reactor = parent.reactor;
+    if (settings.find("collector_base_url") == settings.end()) {
+        // Note: by default we use the testing collector URL because otherwise
+        // testing runs would be collected creating noise and using resources
+        settings["collector_base_url"] = collector::testing_collector_url();
+    }
+    logger->info("Using collector: %s", settings["collector_base_url"].c_str());
+}
+
+/* static */ Var<BaseReporter> OoniReporter::make(const OoniTest &parent) {
```

Comment:
The method is declared static in the class. Here in the implementation I've prefixed a commented out `static` to remind the reader that the method is static.

It's a convention I've seen used somewhere, and thought it was useful as a reminder; now I also see that it can be confusing and I'm less sympathetic towards it than before.